sdk/protocol-plugins/src/aave-v3/AAVEv3ProtocolPlugin.ts

```diff
@@ -7,34 +7,66 @@ import {
   Price,
   CurrencySymbol,
   RiskRatio,
+  ChainFamilyName,
+  valuesOfChainFamilyMap,
+  ChainId,
 } from '@summerfi/sdk-common/common'
 import type { AaveV3PoolId } from '@summerfi/sdk-common/protocols'
 import { PoolType, ProtocolName, EmodeType } from '@summerfi/sdk-common/protocols'
 import { BigNumber } from 'bignumber.js'
 import { z } from 'zod'
+import { BaseProtocolPlugin } from '../implementation/BaseProtocolPlugin'
+import { AaveV3LendingPool, AaveV3PoolCollateralConfig, AaveV3PoolDebtConfig } from './Types'
 import {
-  AaveV3LendingPool,
-  AaveV3PoolCollateralConfig,
-  AaveV3PoolDebtConfig,
-  ChainId,
-  IPositionId,
-  ProtocolManagerContext,
-  ProtocolPlugin,
-} from '../interfaces'
-import { AaveV3LikePluginBuilder, filterAssetsListByEMode } from './AAVEv3LikeBuilder'
-import { UNCAPPED_SUPPLY, PRECISION_BI } from './constants'
-
-export class AaveV3Plugin implements ProtocolPlugin<AaveV3PoolId> {
-  public readonly protocol = ProtocolName.AAVEv3
-  supportedChains = [ChainId.Mainnet, ChainId.Arbitrum, ChainId.Optimism]
-  async getPool(aaveV3PoolId: unknown, ctx: ProtocolManagerContext): Promise<AaveV3LendingPool> {
+  AaveV3LikePluginBuilder,
+  filterAssetsListByEMode,
+} from '../implementation/AAVEv3LikeBuilder'
+import { UNCAPPED_SUPPLY, PRECISION_BI } from '../implementation/constants'
+
+export class AaveV3ProtocolPlugin extends BaseProtocolPlugin<AaveV3PoolId> {
+  public static protocol: ProtocolName.AAVEv3 = ProtocolName.AAVEv3
```

Comment:
We should not use static here, as it forces you to then pass the values to BaseProtocolPlugin in the constructor, which complicates the initialization chain. Instead:

```
abstract class BaseProtocolPlugin<PoolId extends ...> {
  public abstract readonly protocol: ProtocolName
  public abstract readonly supportedChains: ChainInfo[]
  public abstract schema: unknown // I guess it has to be unknown, unless Zod has a base type for this
}

class AaveV3ProtocolPlugin extends BaseProtocolPlugin<AaveV3PoolId> {
  public readonly protocol = ProtocolName.AAVEv3
  public readonly supportedChain = valuesOfChain...
  public readonly schema = z.object...
}
```

Reply:
It's because the 

  isPoolId(candidate: unknown): asserts candidate is PoolIdType {
    const parseResult = this.schema.safeParse(candidate)
    if (!parseResult.success) {
      const errorDetails = parseResult.error.errors
        .map((error) => `${error.path.join('.')} - ${error.message}`)
        .join(', ')
      throw new Error(`Candidate is not correct: ${errorDetails}`)
    }
  }
  
  Is implemented on the BaseClass. 
  
  We can get round it by setting values inside the constructor and passing to the base class there.

Reply:
Can ignore my comment. Static props removed